vpr/src/route/router_lookahead.cpp

```diff
@@ -42,55 +42,60 @@ std::unique_ptr<RouterLookahead> make_router_lookahead(
+
+    if (rr_type == CHANX || rr_type == CHANY) {
+        VTR_ASSERT_SAFE(device_ctx.rr_nodes[node].type() == CHANX || device_ctx.rr_nodes[node].type() == CHANY);
+
+        int num_segs_ortho_dir = 0;
+        int num_segs_same_dir = get_expected_segs_to_target(node, target_node, &num_segs_ortho_dir);
 
-    int num_segs_ortho_dir = 0;
-    int num_segs_same_dir = get_expected_segs_to_target(inode, target_node, &num_segs_ortho_dir);
+        int cost_index = device_ctx.rr_nodes[node].cost_index();
+        int ortho_cost_index = device_ctx.rr_indexed_data[cost_index].ortho_cost_index;
 
-    int cost_index = device_ctx.rr_nodes[inode].cost_index();
-    int ortho_cost_index = device_ctx.rr_indexed_data[cost_index].ortho_cost_index;
+        const auto& same_data = device_ctx.rr_indexed_data[cost_index];
+        const auto& ortho_data = device_ctx.rr_indexed_data[ortho_cost_index];
+        const auto& ipin_data = device_ctx.rr_indexed_data[IPIN_COST_INDEX];
+        const auto& sink_data = device_ctx.rr_indexed_data[SINK_COST_INDEX];
 
-    const auto& same_data = device_ctx.rr_indexed_data[cost_index];
-    const auto& ortho_data = device_ctx.rr_indexed_data[ortho_cost_index];
-    const auto& ipin_data = device_ctx.rr_indexed_data[IPIN_COST_INDEX];
-    const auto& sink_data = device_ctx.rr_indexed_data[SINK_COST_INDEX];
+        float cong_cost = num_segs_same_dir * same_data.base_cost
+                          + num_segs_ortho_dir * ortho_data.base_cost
+                          + ipin_data.base_cost
+                          + sink_data.base_cost;
 
-    float cong_cost = num_segs_same_dir * same_data.base_cost
-                      + num_segs_ortho_dir * ortho_data.base_cost
-                      + ipin_data.base_cost
-                      + sink_data.base_cost;
+        float Tdel = num_segs_same_dir * same_data.T_linear
+                     + num_segs_ortho_dir * ortho_data.T_linear
+                     + num_segs_same_dir * num_segs_same_dir * same_data.T_quadratic
+                     + num_segs_ortho_dir * num_segs_ortho_dir * ortho_data.T_quadratic
+                     + R_upstream * (num_segs_same_dir * same_data.C_load + num_segs_ortho_dir * ortho_data.C_load)
+                     + ipin_data.T_linear;
 
-    float Tdel = num_segs_same_dir * same_data.T_linear
-                 + num_segs_ortho_dir * ortho_data.T_linear
-                 + num_segs_same_dir * num_segs_same_dir * same_data.T_quadratic
-                 + num_segs_ortho_dir * num_segs_ortho_dir * ortho_data.T_quadratic
-                 + R_upstream * (num_segs_same_dir * same_data.C_load + num_segs_ortho_dir * ortho_data.C_load)
-                 + ipin_data.T_linear;
+        return std::make_pair(params.criticality * Tdel, (1 - params.criticality) * cong_cost);
+    } else if (rr_type == IPIN) { /* Change if you're allowing route-throughs */
+        return std::make_pair(device_ctx.rr_indexed_data[SINK_COST_INDEX].base_cost, 0.);
```

Comment:
Should the base_cost be on the second part of this pair?  Congestion is the pair.second?

Reply:
Good catch, I assumed base cost was related to delay, wrongly.